src/store/api/worldUsers.ts

```diff
@@ -0,0 +1,193 @@
+
+import {
+  extractLocationFromUser,
+  User,
+  UserLocation,
+  UserWithLocation,
+  userWithLocationToUser,
+} from "types/User";
+
+import { withId, WithId } from "utils/id";
+
+export interface WorldUsersApiArgs {
+  relatedLocationIds: string[];
+}
+
+export interface WorldUsersData {
+  // TODO: this is basically the naive 'copy/paste' data structure that matches the legacy
+  //   react-redux-firebase useFirestoreConnect implementation currently. Once we have this in
+  //   place/working, we want to refactor it to split the data in a 'smarter'/more efficient way.
+  worldUsers: WithId<User>[];
+  // @debt worldUsersById should be Partial<Record<string, WithId<User>>>, current kept as is for legacy reasons, but it's technically a typing bug
+  worldUsersById: Record<string, WithId<User>>;
+  // @debt worldUserLocationsById should be Partial<Record<string, WithId<UserLocation>>>, current kept as is for legacy reasons, but it's technically a typing bug
+  worldUserLocationsById: Record<string, WithId<UserLocation>>;
+}
+
+export const initialState: Readonly<WorldUsersData> = {
+  worldUsers: [],
+  worldUsersById: {},
+  worldUserLocationsById: {},
+};
+
+// TODO: https://redux-toolkit.js.org/rtk-query/api/createApi
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#streaming-data-with-no-initial-request
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#adding-meta-information-to-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#constructing-a-dynamic-base-url-using-redux-state
+//   Can we use this to read the sovereignVenueId (or even currentVenueId or maybe all relatedVenueIds as needed) directly out of the state?
+//   And even if we can, do we want to? Or keep passing it in through WorldUsersApiArgs? (maybe the latter, as I think then it
+//     is used for/relates to the cache key for this data)
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/migrating-to-rtk-query
+export const worldUsersApi = createApi({
+  reducerPath: "worldUsersApi",
+  baseQuery: fakeBaseQuery(),
+  endpoints: (builder) => ({
+    worldUsers: builder.query<WorldUsersData, WorldUsersApiArgs>({
+      // The query is not relevant here as the data will be provided via streaming updates.
+      // A queryFn returning an empty array is used, with contents being populated via
+      // streaming updates below as they are received.
+      queryFn: () => ({
+        data: initialState,
+        // TODO: we can also return error and/or meta here as well as data. Do we need to be able to?
+        // error: undefined,
+        // meta: undefined,
+      }),
+      async onCacheEntryAdded(
+        { relatedLocationIds },
+        { updateCachedData, cacheEntryRemoved }
+      ) {
+        // TODO: use api/* layer for this?
+        const usersCollectionRef = firebase.firestore().collection("users");
```

Comment:
doesn't seem to be using a ref, so the const name is a bit misleading? might want to rename it

Reply:
It's using a firestore collection ref, not a react ref. This is fine, and matches our naming patterns across the rest of the codebase.